test/jasmine/tests/transforms_test.js

```diff
@@ -63,6 +63,40 @@ describe('one-to-one transforms:', function() {
         expect(traceOut.y).toBe(traceIn.y);
     });
 
+    it('supplyTraceDefaults should honored global transforms', function() {
+        var traceIn = {
+            y: [2, 1, 2],
+            transforms: [{
+                type: 'filter',
+                operation: '>',
+                value: '0',
+                filtersrc: 'x'
+            }]
+        };
+
+        var layout = {
+            _globalTransforms: [{
+                type: 'filter'
+            }]
+        };
+
+        var traceOut = Plots.supplyTraceDefaults(traceIn, 0, layout);
+
+        expect(traceOut.transforms[0]).toEqual({
+            type: 'filter',
+            operation: '=',
+            value: 0,
+            filtersrc: 'x'
+        }, '- global first');
+
+        expect(traceOut.transforms[1]).toEqual({
+            type: 'filter',
+            operation: '>',
+            value: 0,
+            filtersrc: 'x'
+        }, '- trace second');
+    });
+
```

Comment:
looks great! 
What do you think about one additional test that ensures transforms have access to `fullTrace._input` user data? Just in case someone other than yourself decides to refactor one day and accidentally breaks a bunch of userspace transforms.